File: 9.20/oc与swift混编/oc与swift混编/NetworkTool.m
//
//  NetworkTool.m
//  oc与swift混编
//
//  Created by codygao on 16/9/20.
//  Copyright © 2016年 HM. All rights reserved.
//

#import "NetworkTool.h"

@implementation NetworkTool
+(instancetype)sharedtool{
    static NetworkTool *instance;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        instance = [[NetworkTool alloc] init];
    });
    
    return instance;
    
}
@end

File: oc与swift混编/oc与swift混编/NetworkTool.swift
//
//  NetworkTool.swift
//  oc与swift混编
//
//  Created by codygao on 16/9/20.
//  Copyright © 2016年 HM. All rights reserved.
//



import UIKit

class NetworkTool: NSObject {

    //创建单例类
    //创建全局单例类
    static var networktool: NetworkTool = NetworkTool()
    
    static var networktool2: NetworkTool = {
        let tool = NetworkTool()
        //仔设置乞讨属性
        return tool
    }()
    
}
